src/parser/TurtleParser.h

```diff
@@ -117,6 +134,11 @@ class TurtleParser {
   std::string _activePredicate;
   size_t _numBlankNodes = 0;
 
+  // throw an exception annotated with position information
+  [[noreturn]] void raise(std::string_view msg) {
+    throw ParseException(msg, getParsePosition());
+  }
+
```

Comment:
For the various parse errors, it would be nice if the error message would not just be "Iriref", but "Expected Iriref".
It would also be nice to show the line where this happened and indicate the position (not only the number, see below) in the line where it went wrong. For example:

```
wd:Q42 <bla< "does not matter"
       ^                 
```

Reply:
This is an additional feature, which is not related to escaping.
I have opened an issue  to preserve your exact wish and will address this in a separate PR